core/src/main/java/io/grpc/transport/inprocess/InProcessTransport.java

```diff
@@ -93,6 +93,7 @@ public void run() {
         }
       }).start();
     }
+    clientTransportListener.transportReady();
```

Comment:
This calls the listener from within start(). Do we want to permit the re-entrant behavior or run this on another thread?

I think you will need to hold a lock on `this` in order to make sure that the listener is called in a thread-safe manner.